assets/ts/schedule/components/line-diagram/StopPredictions.tsx

```diff
@@ -55,13 +55,24 @@ const StopPredictions = ({
         ? "m-schedule-diagram__cr-prediction-time strikethrough"
         : "m-schedule-diagram__cr-prediction-time";
 
+      let headsignCopy = headsign;
+      if (
+        headsign?.time_data_with_crowding_list[0]?.predicted_schedule?.schedule
+          ?.stop_headsign
+      ) {
+        const headsignName =
+          headsign.time_data_with_crowding_list[0].predicted_schedule.schedule
+            .stop_headsign;
+        headsignCopy = { ...headsign, name: headsignName };
+      }
+
       return (
-        <div key={`${headsign.name}-cr`}>
+        <div key={`${headsignCopy.name}-cr`}>
           <div className="m-schedule-diagram__cr-prediction">
             {timeForCommuterRail(predictedOrScheduledTime, predictionTimeClass)}
           </div>
           <div className="m-schedule-diagram__cr-prediction-details">
-            {predictionTexts(headsign).map((text: string) => (
+            {predictionTexts(headsignCopy).map((text: string) => (
```

Comment:
question: Isn't `headsign.name` already corrected via the backend change? Not sure we need any of this...

Reply:
Unfortunately, it isn't for every case. This is actually effecting three different places rather than two.

1. The line diagram
2. The list of trips in the schedule finder
3. The predictions that sometimes appear at the top of the schedule finder

Reply:
I figured changing it like this is less risky and time-consuming than trying to figure out when and where the backend data is getting overwritten since there are functions that are like...if commuter rail do X, if bus do Y.